api/common/ctx.go

```diff
@@ -30,3 +32,20 @@ func LoggerWithFields(ctx context.Context, fields logrus.Fields) (context.Contex
 	ctx = WithLogger(ctx, l)
 	return ctx, l
 }
+
+// GetRemainingTime extracts the deadline from the context and returns the
+// remaining time left.
+func GetRemainingTime(ctx context.Context) (*time.Duration, error) {
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return nil, errors.New("call without deadline provided")
```

Comment:
is it really possible to get a call without deadline?

Reply:
yea, i think for now it should be impossible but I could see in the future people wanting to fork and remove the timeouts to run really long running tasks (certain users are really stubborn and don't want to shorten their tasks, always happens)